### PR TITLE
Add native syllabus routes

### PR DESCRIPTION
**Pull request by balintbartok**
refs: MBL-14814
affects: Teacher
release note: none

test plan:
* turn on experimental feature: native_teacher_syllabus
* check a course, that has a syllabus
* native syllabus page should open

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student.ipa</summary>
  <a href="https://app.bitrise.io/artifact/53120909/p/550b5f5ff59e2d32fbdb85689e0aed02">
    
  </a>
</details>
<details><summary>Teacher.ipa</summary>
  <a href="https://app.bitrise.io/artifact/53120927/p/46d79ac8e60097283df423756731a109">
    
  </a>
</details>
<details><summary>Parent.ipa</summary>
  <a href="https://app.bitrise.io/artifact/53120874/p/bc42fdf4ae6b287b009276c0759b19be">
    
  </a>
</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


<details><summary>:x: Yarn lint</summary>


    
    /Users/vagrant/git/rn/Teacher/src/modules/courses/CourseNavigation.js
      160:1  error  Expected indentation of 10 spaces but found 12  indent
      161:1  error  Expected indentation of 10 spaces but found 12  indent
    
    ✖ 2 problems (2 errors, 0 warnings)
      2 errors and 0 warnings potentially fixable with the `--fix` option.
    
</details>
<details><summary>:x: XCTest failed: StudentUnitTests/UrlSubmissionPresenterTests/testSubmitSuccess</summary>


    Crash: Student (15728): Namespace SPRINGBOARD, Code 0x8badf00d
    Assertion Failure: Crash: Student (15728): Namespace SPRINGBOARD, Code 0x8badf00d
</details>


[MBL-14814](https://instructure.atlassian.net/browse/MBL-14814)

#### Affected Apps: Teacher
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: rn/Teacher/ios/Teacher/Routes.swift [hunk 1]
```diff
@@ -291,6 +289,15 @@ private func fileEditor(url: URLComponents, params: [String: String], userInfo:
     return CoreHostingController(FileEditorView(context: Context(path: url.path), fileID: fileID))
 }
 
+private func syllabus(url: URLComponents, params: [String: String], userInfo: [String: Any]?) -> UIViewController? {
+    guard ExperimentalFeature.nativeTeacherSyllabus.isEnabled else {
+        Router.open(url: url)
+        return nil
+    }
+    guard let courseID = params["courseID"] else { return nil }
+    return SyllabusViewController.create(courseID: ID.expandTildeID(courseID))
```
Comment (vanwagonet) [inline rn/Teacher/ios/Teacher/Routes.swift:298]: nice remembering to expand IDs with tildes.

File: rn/Teacher/ios/TeacherTests/RoutesTests.swift [hunk 2]
```diff
@@ -91,5 +91,11 @@ class RoutesTests: XCTestCase {
         XCTAssert(router.match("/dev-menu/experimental-features") is ExperimentalFeaturesViewController)
         XCTAssert(router.match("/support/problem") is ErrorReportViewController)
         XCTAssert(router.match("/support/feature") is ErrorReportViewController)
+        ExperimentalFeature.nativeTeacherSyllabus.isEnabled = false
+        XCTAssertNil(router.match( "/courses/1/assignments/syllabus"))
+        XCTAssertNil(router.match( "/courses/1/syllabus"))
+        ExperimentalFeature.nativeTeacherSyllabus.isEnabled = true
+        XCTAssert(router.match( "/courses/1/assignments/syllabus") is SyllabusViewController)
+        XCTAssert(router.match( "/courses/1/syllabus") is SyllabusViewController)
```
Comment (vanwagonet) [inline rn/Teacher/ios/TeacherTests/RoutesTests.swift:99]: good tests.

**Review by vanwagonet — CHANGES_REQUESTED** [review]
mapStateToProps in CourseNavigation isn't including the syllabus tab for teacher, so I can't test if it routes correctly. Teacher will only show tabs with ids in the `availableCourseTabs` array.

File: rn/Teacher/ios/Teacher/Routes.swift (hunk 1 above)
Comment (balintbartok) [reply rn/Teacher/ios/Teacher/Routes.swift:298]: the power of copypaste ;)

**Review by vanwagonet — APPROVED** [review]
QA+1, sorry I missed that change the first time & had tried to test without the flag enabled.

**Pull request merged**